feat: add `abstractify` helper function to `core.operator` module (#9694)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

  [(#9675)](https://github.com/PennyLaneAI/pennylane/pull/9675)
  [(#9685)](https://github.com/PennyLaneAI/pennylane/pull/9685)

> * Added a new `abstractify` utility function that is able to convert various objects

Author: ```suggestion
* Added an internal `abstractify` utility function that is able to convert various objects
```
